apps/cyberstorm-remix/app/root.tsx

```diff
@@ -158,22 +165,23 @@ export type RootLoadersType = typeof loader | typeof clientLoader;
 export function shouldRevalidate({
   defaultShouldRevalidate,
 }: ShouldRevalidateFunctionArgs) {
-  if (defaultShouldRevalidate) return true;
   const publicEnvVariables = getPublicEnvVariables([
     "VITE_API_URL",
     "VITE_COOKIE_DOMAIN",
   ]);
-  if (
-    !sessionValid(
-      new StorageManager(SESSION_STORAGE_KEY),
-      publicEnvVariables.VITE_API_URL || "",
-      publicEnvVariables.VITE_COOKIE_DOMAIN || ""
-    )
-  )
-    return true;
-  return getSessionStale(new NamespacedStorageManager(SESSION_STORAGE_KEY));
+  sessionValid(
+    new StorageManager(SESSION_STORAGE_KEY),
+    publicEnvVariables.VITE_API_URL || "",
+    publicEnvVariables.VITE_COOKIE_DOMAIN || ""
+  );
+  const sessionIsStale = getSessionStale(
+    new NamespacedStorageManager(SESSION_STORAGE_KEY)
+  );
+  if (sessionIsStale || defaultShouldRevalidate) return true;
```

Comment:
Minor: `return sessionIsStale || defaultShouldRevalidate;` would be better as it's less to read and would change the functions return type from awkward `true | undefined` to `boolean`.